### PR TITLE
update documentation

### LINKED ISSUE #166: Include details of 2.0 changes in CHANGELOG/README
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

If the .conda has info- as its last file `unzip -l x.conda`, it was produced by version 2. Those files will also have the minimum timestamp for ZIP. So far conda-forge isn’t producing those yet (I looked at fractopo-0.5.1-pyhd8ed1ab_0.conda). The embedded tarfiles no longer include uid/gid/username/groupname, but they still include the current timestamp.

Follow up question should we support a SOURCE_DATE_EPOCH for the timestamps, instead of going with what the filesystem said

### Conda Info

_No response_

### Conda Config

_No response_

### Conda list

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dholth**
### Description

Various documentation updates, to highlight 2.0 and provide better cli documentation.

Fixes #166 

Is there a more elegant way to deal with "include README in sphinx docs"? I might like to pare README down to a one-line description and a link to the sphinx docs.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: .github/workflows/sphinx.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ jobs:
       - name: Build Documentation
         run: |
           pip install -e .[docs]
+          pip install -r docs/requirements.txt  # redundant?
```
Comment (jezdez) [inline .github/workflows/sphinx.yml:26]: Yep redundant

**Review by jezdez — APPROVED** [review]
NICE <3

File: .github/workflows/sphinx.yml (hunk 1 above)
Comment (dholth) [reply .github/workflows/sphinx.yml:26]: The full question would be: does the sphinx build automatically find docs/requirements.txt

Unfortunately there isn't a good way to make sure our pip "docs" extra matches docs/requirements.txt

File: README.md [hunk 2]
```diff
@@ -1,101 +1,8 @@
...
-
-If for some reason a release fails partway through, or you want to claw back a
-release that you have made, rever allows you to undo activities. If you find
-yourself in this pickle, you can pass the ``--undo`` option a comma-separated
-list of activities you'd like to undo.  For example::
-
-     rever --undo tag,changelog,authors 1.2.3
-
- Happy releasing!
+Full documentation at [https://conda.github.io/conda-package-handling/](https://conda.github.io/conda-package-handling/)
```
Comment (dholth) [inline README.md:8]: Maybe later we can add a standard "here are all the project links" section here.

**Pull request merged**